src/core/blockchain/ethereum/client.ts

```diff
@@ -160,7 +162,7 @@ export class Client extends BlockchainGenericClient {
             this.http.jsonRpc('eth_getTransactionReceipt', transactionHash)
         ];
 
-        return Promise.all(rpcCalls).then(res => {
+        return Promise.all(rpcCalls).then(async res => {
```

Comment:
all method from clients should be kept as light as we can, it should only facilitate network requests. For data transformation create a function in `core/${blockchain}/transaction.ts` for this particular case: `buildTransactionFromBlockchainData(blockchainData): IBlockchainTransaction<...>`

Reply:
refactore and moved this in client-utils.ts